tests/overmap_test.cpp

```diff
@@ -19,3 +21,30 @@ TEST_CASE( "set_and_get_overmap_scents" ) {
     REQUIRE( test_overmap.scent_at( { 75, 85, 0} ).creation_turn == 50 );
     REQUIRE( test_overmap.scent_at( { 75, 85, 0} ).initial_strength == 90 );
 }
+
+TEST_CASE( "default_overmap_generation_always_succeeds" ) {
+    // List of specials to generate, overmap should consume them all.
+    std::vector<const overmap_special *> default_specials = overmap( 0, 0 ).get_enabled_specials();
+    int overmaps_to_construct = 10;
+    for( point candidate_addr : closest_points_first( 10, { 0, 0 } ) ) {
+         // Skip populated overmaps.
+      if( overmap_buffer.has( candidate_addr.x, candidate_addr.y ) ) {
+            continue;
+        }
+        std::vector<overmap_special_placement> test_specials;
+        for( auto special : default_specials ) {
+            test_specials.push_back( { 0, special } );
+        }
+        overmap_buffer.create_custom_overmap( candidate_addr.x, candidate_addr.y, test_specials );
```

Comment:
Each iteration through this loop is essentially independent, they just happen in the same game world because generating a new one is super slow.  That having been said, this code excersises the logic for scanning for non-existent overlaps and uncovered some bugs related to that.  In pseudocode:

```
Until we've placed 10 sets of specials:
    Find an adjacent and uninitialized overmap location
        Construct an overmap there using a full list of specials
            Recursively find nearby overmap positions
                generate overmaps until mandatory specials are all placed.
        Assert that list of specials has been exausted of mandatory specials.
```
More tersely, the specials list isn't reset until we've successfully asserted that it generated all of its specials.